pkg/api/v1/resource/helpers_test.go

```diff
@@ -66,48 +74,93 @@ func TestDefaultResourceHelpers(t *testing.T) {
+			pod:           getPod(containerName, resources{cpuOverhead: "5", memoryOverhead: "5"}),
 			resourceName:  v1.ResourceCPU,
 			expectedValue: 0,
 		},
 		{
 			cName:         "just-overhead for memory",
-			pod:           getPod("foo", podResources{memoryOverhead: "5"}),
+			pod:           getPod(containerName, resources{memoryOverhead: "5"}),
 			resourceName:  v1.ResourceMemory,
 			expectedValue: 0,
 		},
 		{
 			cName:         "cpu overhead and req",
-			pod:           getPod("foo", podResources{cpuRequest: "2", cpuOverhead: "5", memoryOverhead: "5"}),
+			pod:           getPod(containerName, resources{cpuRequest: "2", cpuOverhead: "5", memoryOverhead: "5"}),
 			resourceName:  v1.ResourceCPU,
 			expectedValue: 7000,
 		},
 		{
 			cName:         "mem overhead and req",
-			pod:           getPod("foo", podResources{cpuRequest: "2", memoryRequest: "1024", cpuOverhead: "5", memoryOverhead: "5"}),
+			pod:           getPod(containerName, resources{cpuRequest: "2", memoryRequest: "1024", cpuOverhead: "5", memoryOverhead: "5"}),
 			resourceName:  v1.ResourceMemory,
 			expectedValue: 1029,
 		},
+		{
+			cName:                    "pod level resources cpu req, container cpu req",
+			pod:                      getPodWithPodLevelResources(containerName, resources{cpuRequest: "10"}, resources{cpuRequest: "8"}),
+			resourceName:             v1.ResourceCPU,
+			expectedValue:            10000,
+			podLevelResourcesEnabled: true,
+		},
+		{
+			cName:                    "pod level resources mem req, container mem req",
+			pod:                      getPodWithPodLevelResources(containerName, resources{memoryRequest: "100Mi"}, resources{memoryRequest: "80Mi"}),
+			resourceName:             v1.ResourceMemory,
+			expectedValue:            104857600,
+			podLevelResourcesEnabled: true,
+		},
+		{
+			cName:                    "pod level resources cpu req, container no req",
+			pod:                      getPodWithPodLevelResources(containerName, resources{cpuRequest: "10"}, resources{}),
+			resourceName:             v1.ResourceCPU,
+			expectedValue:            10000,
+			podLevelResourcesEnabled: true,
+		},
+		{
+			cName:                    "pod level resources mem req, container no req",
+			pod:                      getPodWithPodLevelResources(containerName, resources{memoryRequest: "100Mi"}, resources{}),
+			resourceName:             v1.ResourceMemory,
+			expectedValue:            104857600,
+			podLevelResourcesEnabled: true,
+		},
+		{
+			cName:                    "cpu pod without pod level resources but pod level resources feature enabled",
+			pod:                      getPod(containerName, resources{cpuRequest: "9"}),
+			resourceName:             v1.ResourceCPU,
+			expectedValue:            9000,
+			podLevelResourcesEnabled: true,
+		},
+		{
+			cName:                    "mem pod without pod level resources but pod level resources feature enabled",
```

Comment:
can we also add a test for memory set at pod level and cpu set at container level and vice versa

Reply:
Done, thank you.

Reply:
Added `pod level resources cpu req, container mem req` and `pod level resources cpu req, container mem req`.